### PR TITLE
EOS-18521:S3:Upgrade bazel version and generate new bazel rpm

### PR DESCRIPTION
**Pull request by sachitanands**
Bazel releases : https://github.com/bazelbuild/bazel/releases/ 
**current version** : 0.13.0
**new version** : 4.1.0

1) Upgrade bazel version 
2) Generate new bazel rpm
3) Update 3rd party repo with new bazel rpm

Added two flags to each bazel command which will prevent gcc compiler kill error 
--local_cpu_resources=HOST_CPUS*.5
--local_ram_resources=HOST_RAM*.5
Reference doc : https://docs.bazel.build/versions/main/command-line-reference.html#flag--local_cpu_resources 

Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

### CODE REVIEW

File: rebuildall.sh [hunk 1]
```diff
@@ -371,32 +371,33 @@ then
...
-
+cpu_resource_limit_param="--local_cpu_resources=HOST_CPUS*.5"
+ram_resource_limit_param="--local_ram_resources=HOST_RAM*.5"
 if [ $no_s3ut_build -eq 0 ]
 then
   bazel build //:s3ut --cxxopt="-std=c++11" --define $MOTR_INC_ \
                       --define $MOTR_LIB_ --define $MOTR_HELPERS_LIB_ \
                       --spawn_strategy=standalone \
-                      --strip=never
+                      --strip=never $cpu_resource_limit_param $ram_resource_limit_param
```
Comment (knrajnambiar76) [inline rebuildall.sh:381]: you can have that within "" , basically what codacy is complaining

**Review by knrajnambiar76 — APPROVED** [review]

File: rebuildall.sh (hunk 1 above)
Comment (sachitanands) [reply rebuildall.sh:381]: yes. will do it

**Review by nileshgovande — APPROVED** [review]

**Comment by sachitanands** [comment]
Tested this code on below vm configurations,
1) CPU -8 , RAM -16GB
2) CPU -4 , RAM -8GB
3) CPU -4 , RAM -4GB
Bazel build went fine and jenkins job completed without any error.

**Review by SeagateChaDeepak — APPROVED** [review]
+1 LGTM

**Pull request merged**